ivadomed/main.py

```diff
@@ -738,62 +694,77 @@ def cmd_test(context):
                     rdict[k] = batch[k][smp_idx]
                 if rdict["input"].shape[0] > 1:
                     rdict["input"] = rdict["input"][1,][None,]
+
+                # If multiclass merge labels
                 rdict_undo = val_undo_transform(rdict)
 
-                fname_ref = rdict_undo['input_metadata']['gt_filename']
+                fname_ref = rdict_undo['gt_metadata']['gt_filenames'][0]
                 if not context['unet_3D']:
                     if pred_tmp_lst and (fname_ref != fname_tmp or (
                             i == len(test_loader) - 1 and smp_idx == len(batch['gt']) - 1)):  # new processed file
                         # save the completely processed file as a nifti file
                         fname_pred = os.path.join(path_3Dpred, fname_tmp.split('/')[-1])
-                        fname_pred = fname_pred.split(context['target_suffix'])[0] + '_pred.nii.gz'
+                        fname_pred = fname_pred.split(context['target_suffix'][0])[0] + '_pred.nii.gz'
 
                         # If MonteCarlo, then we save each simulation result
                         if n_monteCarlo > 1:
                             fname_pred = fname_pred.split('.nii.gz')[0] + '_' + str(i_monteCarlo).zfill(2) + '.nii.gz'
 
-                        _ = pred_to_nib(data_lst=pred_tmp_lst,
-                                        z_lst=z_tmp_lst,
-                                        fname_ref=fname_tmp,
-                                        fname_out=fname_pred,
-                                        slice_axis=AXIS_DCT[context['slice_axis']],
-                                        kernel_dim='2d',
-                                        bin_thr=0.5 if context["binarize_prediction"] else -1)
+                        output_nii = pred_to_nib(data_lst=pred_tmp_lst,
+                                                 z_lst=z_tmp_lst,
+                                                 fname_ref=fname_tmp,
+                                                 fname_out=fname_pred,
+                                                 slice_axis=AXIS_DCT[context['slice_axis']],
+                                                 kernel_dim='2d',
+                                                 bin_thr=0.5 if context["binarize_prediction"] else -1)
+
+                        if output_nii.get_data().shape[-1] > 1:
+                            save_color_labels(output_nii.get_data(),
+                                              context["binarize_prediction"],
+                                              fname_tmp,
+                                              fname_pred.split(".nii.gz")[0] + '_color.nii.gz',
+                                              AXIS_DCT[context['slice_axis']])
```

Comment:
In the case of multiclass --> is the prediction saved as one or multiple files? If one file --> it will be seen with different color when using fsleyes & cie anyway, no? same for multiple files. I may miss something :(

Reply:
The labels are saved as one file. If the number of class is different than 3 (which can be seen as an RGB image only if there is 3 channels in last dimension), the file cannot be seen (to the best of my knowledge) as RGB. For instance, I tried with 2 labels only (tumor and edema), the last dimension is 2 -> I cannot use the RGB view option.I can only assume the same if there were 4 or more labels since the viewer cannot interprete what color you want on an image not encoded in RGB. For multiple files, a few manual manipulations are needed to see each labels in different color. I find it more convenient to see my result as one file :). Does it make sense?

Reply:
I am not sure to know which RGB view option you are referring to.
I was thinking of something like:
```
cd spinalcordtoolbox/data/PAM50/template/
fsleyes PAM50_levels.nii.gz -cm subcortical
```
or another colormap.

Reply:
But a very good application for the way you implemented it --> is to be able to manipulate it directly in Python / matplotlib etc which could be very handy. Same idea with [data_painted](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/utils.py#L265) from eval.